lua/nvim-treesitter/highlight.lua

```diff
@@ -0,0 +1,19 @@
+local api = vim.api
+local queries = require'nvim-treesitter.query'
+local ts = vim.treesitter
+
+local M = {}
+
+function M.setup(bufnr, ft)
+  local buf = bufnr or api.nvim_get_current_buf()
+  local ft = ft or api.nvim_buf_get_option(buf, 'ft')
+
+  local query = queries.get_query(ft, "highlights")
+  if not query then return end
+
+  -- Remove default syntax highlighting
+  api.nvim_buf_set_option(buf, 'syntax', 'OFF')
+  ts.TSHighlighter.new(query, buf, ft)
```

Comment:
do we need to attach to the buffer so we can close the highlighter when leaving the buffer ?

Reply:
Hmmm yeah that's would be great actuallly, I'll add that.